sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_async_lro.py

```diff
@@ -73,3 +76,93 @@ async def _poll(self):  # pylint:disable=invalid-overridden-method
                 self._pipeline_response.http_response
             )
+
+class AnalyzeHealthcareEntitiesAsyncLROPollingMethod(TextAnalyticsAsyncLROPollingMethod):
+
+    @property
+    def _current_body(self):
+        from ._generated.v3_1_preview_3.models import JobMetadata
+        return JobMetadata.deserialize(self._pipeline_response)
+
+    @property
+    def created_on(self):
+        if not self._current_body:
+            return None
+        return self._current_body.created_date_time
+
+    @property
+    def expires_on(self):
+        if not self._current_body:
+            return None
+        return self._current_body.expiration_date_time
+
+    @property
+    def last_modified_on(self):
+        if not self._current_body:
+            return None
+        return self._current_body.last_update_date_time
+
+    @property
+    def id(self):
+        if not self._current_body:
+            return None
+        return self._current_body.job_id
+
+
+class AnalyzeHealthcareEntitiesAsyncLROPoller(AsyncLROPoller[PollingReturnType]):
+
+    @property
+    def created_on(self):
+        return self._polling_method.created_on
+
+    @property
+    def expires_on(self):
+        return self._polling_method.expires_on
+
+    @property
+    def last_modified_on(self):
+        return self._polling_method.last_modified_on
+
+    @property
+    def id(self):
+        return self._polling_method.id
+
+    async def cancel( # type: ignore
+        self,
+        **kwargs
+    ):
+        """Cancel the operation currently being polled.
+
+        :keyword int polling_interval: The polling interval to use to poll the cancellation status.
+            The default value is 5 seconds.
+        :return: Returns an instance of an LROPoller that returns None.
+        :rtype: ~azure.core.polling.LROPoller[None]
+        :raises: Warning when the operation has already reached a terminal state.
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/async_samples/sample_analyze_healthcare_entities_with_cancellation_async.py
+                :start-after: [START analyze_healthcare_entities_with_cancellation_async]
+                :end-before: [END analyze_healthcare_entities_with_cancellation_async]
+                :language: python
+                :dedent: 8
+                :caption: Cancel an existing health operation.
+        """
+        polling_interval = kwargs.pop("polling_interval", 5)
+
+        terminal_states = ["cancelled", "cancelling", "failed", "succeeded", "partiallyCompleted", "rejected"]
+        await self._polling_method.update_status()
+
+        if self._polling_method.status() in terminal_states:
+            raise Warning("Operation with ID '%s' is already in a terminal state and cannot be cancelled." \
+                % self.id)
+
+        try:
+            return await getattr(self._polling_method, "_client").begin_cancel_health_job(
```

Comment:
tiny nit: you can just do self._polling_method._client instead of using getattr

Reply:
I tried this, but the linter was complaining about this since I guess the base `LROPoller` object is not strictly guaranteed to have these properties.  I'm not sure how to silence this particular issue using the linter directives.

Reply:
oh that's really interesting. yeah i guess just leave it for now